pytorch_lightning/trainer/properties.py

```diff
@@ -264,20 +336,87 @@ def checkpoint_callbacks(self) -> List[ModelCheckpoint]:
     def save_checkpoint(self, filepath, weights_only: bool = False):
         self.checkpoint_connector.save_checkpoint(filepath, weights_only)
 
+    @property
+    def model(self) -> Any:
+        """
+        The LightningModule, but possibly wrapped into DataParallel or DistributedDataParallel.
+        To access the pure LightningModule, use
+        :meth:`~pytorch_lightning.trainer.trainer.Trainer.lightning_module` instead.
+        """
+        return self.accelerator.model
+
+    @model.setter
+    def model(self, model: Any):
```

Comment:
Can we add a setter in the training_type_plugin. So if the user provides a LightningModule, we wrap for into in DDP, DP, etc...

Reply:
Actually this was done on purpose. If we actually need to set something there, we might have issues in our control flow, meaning that we then have issues there. So if something is set there, it should either error out or people know what they're doing and do it on purpose and then use the protected attribute.

Reply:
the setter here is only used for the tuner (lr find, batch size scaler etc.) at the moment and generally I would prefer not to let the user mess with the model this way. If the user wants to add a wrapper they should do that before passing the model to fit.